src/GenericParameters.cc

```diff
@@ -17,7 +17,7 @@ GenericParameters::GenericParameters(const GenericParameters& other) :
     m_stringMtx(std::make_unique<std::mutex>()),
     m_doubleMtx(std::make_unique<std::mutex>()) {
   {
-    // acquire all three locks at once to make sure all three internal maps are
+    // acquire all four locks at once to make sure all four internal maps are
```

Comment:
```suggestion
    // acquire all locks at once to make sure all internal maps are
```
What about this because it's likely if the number is changed the comments will not be changed?

Reply:
I would also be fine with that yes :)